### PR TITLE
Quieter logs in CI

### PR DESCRIPTION
**Pull request by clee2000**
To reduce the amount of logs
* for successes, only print the part that says what tests ran and don't print the rest.  Zip the log into an artifact.  The line listing al the test names is really long, but if you view source of the raw logs, it will not wrap so it will only be one line.  The log classifier can also be configured to ignored this line. Gets rid of lines like `test_ops.py::TestCommonCPU::test_multiple_devices_round_cpu_int64 SKIPPED [0.0010s] (Only runs on cuda) [  9%]`
* for failures/reruns, print logs.  Do not zip.

Also
* change log artifact name 

Examples of various logs: 
https://hud.pytorch.org/pytorch/pytorch/commit/a074db0f7f27390ebacb4b5b198a990e2016b194 failures
https://hud.pytorch.org/pytorch/pytorch/commit/1b439e24c42b8fac4ff6e3c1c0b706baec2e319f failures

possibly controversial haha
should i include an option for always printing?

### CODE REVIEW

File: test/run_test.py [hunk 1]
```diff
@@ -920,33 +920,38 @@ def run_doctests(test_module, test_directory, options):
...
-def print_log_file(test: str, file_path: str, failed: bool) -> None:
-    num_lines = sum(1 for _ in open(file_path, "rb"))
+def sanitize_file_name(file: str):
+    return file.replace("\\", ".").replace("/", ".").replace(" ", "_")
+
+
+def handle_log_file(
+    test: ShardedTest, file_path: str, failed: bool, was_rerun: bool
+) -> None:
     test = str(test)
```
Comment (ZainRizvi) [inline test/run_test.py:930]: Should this be test.name, now that test is a ShardedTest?

File: test/test_ops.py [hunk 2]
```diff
@@ -466,6 +466,7 @@ def _distance(a, b):
     @ops(python_ref_db)
     @skipIfTorchInductor("Takes too long for inductor")
     def test_python_ref(self, device, dtype, op):
+        self.assertEqual(1, 2)
```
Comment (ZainRizvi) [inline test/test_ops.py:469]: good luck :P

File: .github/actions/upload-test-artifacts/action.yml [hunk 3]
```diff
@@ -43,14 +43,14 @@ runs:
         FILE_SUFFIX: ${{ inputs.file-suffix }}
       run: |
         # Remove any previous test reports if they exist
-        rm -f usage-log-*.zip
```
Comment (ZainRizvi) [inline .github/actions/upload-test-artifacts/action.yml:46]: Why does this change?

**Review by ZainRizvi — COMMENTED** [review]
(sorry if I'm reviewing this too early)

Mind including some examples of what you expect the logs to look like before/after this change?

File: test/run_test.py (hunk 1 above)
Comment (clee2000) [reply test/run_test.py:930]: str(test) also includes stuff about which chunk of the file its running

File: .github/actions/upload-test-artifacts/action.yml (hunk 3 above)
Comment (clee2000) [reply .github/actions/upload-test-artifacts/action.yml:46]: it doesn't have to change, i just think it's a more appropriate name

File: test/run_test.py [hunk 4]
```diff
@@ -920,33 +920,38 @@ def run_doctests(test_module, test_directory, options):
     return result
 
 
-def print_log_file(test: str, file_path: str, failed: bool) -> None:
-    num_lines = sum(1 for _ in open(file_path, "rb"))
+def sanitize_file_name(file: str):
+    return file.replace("\\", ".").replace("/", ".").replace(" ", "_")
+
+
+def handle_log_file(
```
Comment (huydhn) [inline test/run_test.py:927]: I think let's keep this function to CI only and leave the log printing as it is when devs use `run_test` locally.  Any thoughts?

File: test/run_test.py [hunk 5]
```diff
@@ -920,33 +928,38 @@ def run_doctests(test_module, test_directory, options):
...
+    test: ShardedTest, file_path: str, failed: bool, was_rerun: bool
+) -> None:
     test = str(test)
-    n = 100
     with open(file_path, "rb") as f:
+        full_text = f.read().decode("utf-8", errors="ignore")
+
+    if not failed and not was_rerun and "=== RERUNS ===" not in full_text:
```
Comment (ZainRizvi) [inline test/run_test.py:942]: Do we do reruns for C++ tests as well?

**Review by ZainRizvi — COMMENTED** [review]
Two requests:
1. Can you please confirm that this won't break our existing log parser rules?
2. Could you share an example of log lines that this is expected to eliminate?

File: test/run_test.py (hunk 5 above)
Comment (clee2000) [reply test/run_test.py:942]: I don't think so but cc @huydhn to confirm

**Comment by clee2000** [comment]
> Two requests:
> 
> 1. Can you please confirm that this won't break our existing log parser rules?
> 2. Could you share an example of log lines that this is expected to eliminate?

1. all failures and reruns should still be printed.  Only successes are cut out, so it should be fine. 
2. Added to PR body

File: test/run_test.py (hunk 5 above)
Comment (huydhn) [reply test/run_test.py:942]: We actually have stepcurrent retry for C++ test.  That plugin works with pytest-cpp last time I check.  I don't have an actual example of C++ test retrying at hand (they are pretty reliable), but the successful C++ tests output is truncated in the same way here from what I see.

**Review by huydhn — APPROVED** [review]
I double check the console log output and it looks good

**Pull request closed without merging**